skimage/filters/tests/test_gaussian.py

```diff
@@ -67,4 +67,14 @@ def test_4d_ok():
     img = np.zeros((5,) * 4)
     img[2, 2, 2, 2] = 1
     res = gaussian(img, 1, mode='reflect')
-    assert np.allclose(res.sum(), 1)
+    assert np.allclose(res.sum(), 1)
+
+
+def test_output_type():
+    img = np.arange(16, dtype=np.uint8).reshape((4, 4))
+    output_type = np.uint8
+    gaussian_img = gaussian(img, 1, output=output_type)
+    assert gaussian_img.dtype == output_type
+    output_image = np.zeros_like(img, dtype=output_type)

```

Comment:
should you also test the cool logic you gave about perserve range and floats / integer output dtypes?

maybe output as int32 or uint32 instead of the same type of dtype for those?